File: solutions/elixir/log-level/1/lib/log_level.ex
defmodule LogLevel do
  def to_label(level, legacy?) do
    cond do
      level == 0 -> if legacy? do :unknown else :trace end
      level == 1 -> :debug
      level == 2 -> :info
      level == 3 -> :warning
      level == 4 -> :error
      level == 5 -> if legacy? do :unknown else :fatal end
      true -> :unknown
    end
  end

  def alert_recipient(level, legacy?) do
    case to_label(level, legacy?) do
      x when x in [:error, :fatal] -> :ops
      :unknown -> if legacy? do :dev1 else :dev2 end
      _ -> :false
    end
  end
end

File: solutions/java/logs-logs-logs/1/src/main/java/LogLevel.java
public enum LogLevel {
    TRACE(1),
    DEBUG(2),
    INFO(4),
    WARNING(5),
    ERROR(6),
    FATAL(42),
    UNKNOWN(0);

    private final int logLevel;


    LogLevel(int logLevel) {
        this.logLevel = logLevel;
    }

    public int getLogLevel() {
        return logLevel;
    }

     public static LogLevel convert(String shortening) {
        return switch (shortening) {
            case "TRC" -> TRACE;
            case "DBG" -> DEBUG;
            case "INF" -> INFO;
            case "WRN" -> WARNING;
            case "ERR" -> ERROR;
            case "FTL" -> FATAL;
            default -> UNKNOWN;
        };
    }

}
